drivers/gles2/shader_gles2.cpp

```diff
@@ -1109,3 +1107,11 @@ ShaderGLES2::ShaderGLES2() {
 ShaderGLES2::~ShaderGLES2() {
 	finish();
 }
+
+bool ShaderGLES2::is_dirty() const {
+	if (active != this || !version || new_conditional_version.key != conditional_version.key) {
+		return true;
+	}
+
+	return false;
+}
```

Comment:
Maybe you could just have:
```
return (active != this || !version || new_conditional_version.key != conditional_version.key);
```
(Just a very little thing in what looks like a really great PR, can't wait to see the speedup)

Reply:
Thanks for catching this code smell, I'll clean it up.